### PR TITLE
DataFrame.values impl

### PR DESCRIPTION
**Pull request by akharche**

### CODE REVIEW

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 1]
```diff
@@ -101,6 +102,85 @@ def hpat_pandas_df_index_impl(df):
...
+    for column in df.data:
+        if isinstance(column, StringArrayType):
+            raise TypingError(f'{func_name}: String type is currently unsupported')
+
+    numba_common_dtype = find_common_dtype_from_numpy_dtypes([column.dtype for column in df.data], [])
+
+    def hpat_pandas_df_values_impl(df):
+        df_values = []
+        row_len = len(get_dataframe_data(df, 0))
+        local_data = literal_unroll(df._data)
```
Comment (AlexanderKalistratov) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: Wow
Never thought about using it this way

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 2]
```diff
@@ -101,6 +102,85 @@ def hpat_pandas_df_index_impl(df):
...
+        column_len = len(local_data)
+
+        for i in prange(row_len):
+            row = numpy.empty(column_len, numba_common_dtype)
+            for j in range(column_len):
+                row[j] = numpy.array(local_data[j])[i]
+            df_values.append(row)
```
Comment (AlexanderKalistratov) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: We can't append to list in prange.
Why do you need a list at all?

File: sdc/tests/test_dataframe.py [hunk 3]
```diff
@@ -512,25 +512,50 @@ def test_impl(df, arr):
...
+        np.random.seed(0)
+        A = np.ones(n)
+        B = np.random.ranf(n)
+
+        for values in values_to_test:
+            with self.subTest(values=values):
+                df = pd.DataFrame({'A': A, 'B': B, 'C': values})
+                self._test_df_values_unboxing(df)
+
+    def _test_df_values(self, df):
```
Comment (densmirn) [inline sdc/tests/test_dataframe.py]: Sorry, but where did you use the method?

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 4]
```diff
@@ -101,6 +101,128 @@ def hpat_pandas_df_index_impl(df):
...
+    Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_df_values*
+    Parameters
+    -----------
+    df: :obj:`pandas.DataFrame`
+       input arg
+    Returns
+    -------
+    :obj: `numpy.ndarray`
+       return a Numpy representation of the DataFrame
+    """
```
Comment (AlexanderKalistratov) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py]: Does pandas support Strings? If so, could you please add limitation section, that only numbers are supported?

Comment (akharche) [reply sdc/datatypes/hpat_pandas_dataframe_functions.py]: Yes, Pandas supports Strings and mixed values types

**Comment by AlexanderKalistratov** [comment]
@akharche could you please rebase this PR?

**Pull request merged**